src/node_crypto.cc

```diff
@@ -1865,6 +1866,30 @@ void SSLWrap<Base>::GetPeerCertificate(
 }
 
 
+template <class Base>
+void SSLWrap<Base>::GetCertificate(
+    const FunctionCallbackInfo<Value>& args) {
+  Base* w;
+  ASSIGN_OR_RETURN_UNWRAP(&w, args.Holder());
+  Environment* env = w->ssl_env();
+
+  ClearErrorOnReturn clear_error_on_return;
+
+  Local<Object> result;
+
+  X509Pointer cert(SSL_get_certificate(w->ssl_.get()));
+  if (!cert)
+    goto done;
+
+  result = X509ToObject(env, cert.get());
+
+ done:
+  if (result.IsEmpty())
+    result = Object::New(env->isolate());
```

Comment:
It's slightly faster to do this in JS land. Just return the empty handle (translates to `undefined`) and do `this._handle.getCertificate() || {}` in JS.